src/bionics.cpp

```diff
@@ -639,6 +780,275 @@ void player::power_bionics()
+        if( action == "HELP_KEYBINDINGS" ) {
+            wrefresh( w_bionics );
+
+        } else if( action == "UP" && content.size() > 1 ) {
+            do {
+                if( cursor == 0 ) {
+                    cursor = static_cast<int>( content.size() ) - 1;
+                    scroll_position = max_scroll_position;
+                } else {
+                    cursor--;
+                }
+                if( scroll_position > 0 && cursor - scroll_position < getmaxy( w_bio_list ) / 2 ) {
+                    scroll_position--;
+                }
+            } while ( content[cursor].second == INT_MAX );
+
+        } else if( action == "DOWN" && content.size() > 1 ) {
+            do {
+                cursor++;
+                if( cursor >= static_cast<int>( content.size() ) ) {
+                    cursor = 0;
+                    scroll_position = 0;
+                }
+                if( scroll_position < max_scroll_position &&
+                    cursor - scroll_position > getmaxy( w_bio_list ) / 2 ) {
+                    scroll_position++;
+                }
+            } while ( content[cursor].second == INT_MAX );
+
+        } else if( action == "LEFT" ) {
+            ( tab_index > 0 ) ? tab_index-- : tab_index = tab_count - 1;
+            cursor = 1;
+            scroll_position = 0;
+            recalc = true;
+
+        } else if( action == "RIGHT" ) {
+            tab_index = ( tab_index + 1 ) % tab_count;
+            cursor = 1;
+            scroll_position = 0;
+            recalc = true;
+
+        } else if( action == "REASSIGN" ) {
+            const long ch = my_bionics[content[cursor].second].invlet;
+            bionic *tmp = bionic_by_invlet( ch );
+            if( tmp == nullptr ) {
+                // Selected an non-existing bionic (or escape, or ...)
+                continue;
+            }
+
+            const long newch = popup_getkey( _( "%s; enter new letter." ),
+                                             bionics[tmp->id].name.c_str() );
+            wrefresh( w_bionics );
+            if( newch == ch || newch == ' ' || newch == KEY_ESCAPE ) {
+                continue;
+            }
+            if( !bionic_chars.valid( newch ) ) {
+                popup( _( "Invalid bionic letter. Only those characters are valid:\n\n%s" ),
+                       bionic_chars.get_allowed_chars().c_str() );
+                continue;
+            }
+            bionic *otmp = bionic_by_invlet( newch );
+            if( otmp != nullptr ) {
+                std::swap( tmp->invlet, otmp->invlet );
+            } else {
+                tmp->invlet = newch;
+            }
+
+        } else if( action == "REMOVE" ) {
+            if( uninstall_bionic( my_bionics[content[cursor].second].id ) ) {
+                recalc = true;
+                continue;
+            }
+
+        } else if( action == "CONFIRM" ) {
+            bionic *tmp = &my_bionics[content[cursor].second];
+            if( tmp == nullptr ) {
```

Comment:
This condition can never be true, you just took the address of something (using the `&` operator). Why not change change `tmp` to be a reference into `my_bionics` and remove the check?